tests/vmcheck/multitest.py

```diff
@@ -0,0 +1,104 @@
+#!/bin/env python
+
+import os
+import sys
+import glob
+import subprocess
+
+
+def main():
+
+    failed = False
+    hosts = []
+    for host in sys.argv[1:]:
+        hosts.append(Host(host))
+
+    for test in glob.iglob(os.path.join(sys.path[0], "test-*.sh")):
+        host = wait_for_next_available_host(hosts)
+        rc = host.flush()
+        failed = failed or rc != 0
+        host.dispatch(test)
+
+    for host in hosts:
+        rc = host.flush()
+        failed = failed or rc != 0
+
+    # fetch the journal from all the hosts which had a failure
+    fetcher = os.path.join(sys.path[0], "fetch-journal.sh")
+    for host in hosts:
+        if host.saw_fail:
+            fetcher_env = dict(os.environ)
+            fetcher_env.update({'VM': host.hostname,
+                                'JOURNAL_LOG':
+                                "vmcheck/%s.journal.log" % host.hostname})
+            subprocess.check_call([fetcher], env=fetcher_env)
+
+    return 1 if failed else 0
+
+
+def wait_for_next_available_host(hosts):
+    while True:
+        for host in hosts:
+            if host.is_done():
+                return host
+        os.wait()
+
+
+class Host:
+
+    def __init__(self, hostname):
+        self.hostname = hostname
+        self.test = ""
+        self._p = None
+        self.saw_fail = False
+
+    def is_done(self):
+        if not self._p:
+            return True
+        return self._p.poll() is not None
+
+    def dispatch(self, test):
```

Comment:
I'm thinking this code is something we could convince e.g. GNU `parallel` (or one of [the clones](https://github.com/mmstick/parallel) into doing for us, but...eh.  I only kind of learned the parallel mini-language one time.